rasa/multi_skill.py

```diff
@@ -0,0 +1,148 @@
+            config: Path to the root configuration file in the project directory.
+
+        Returns:
+            `SkillSelector` which specifies the loaded skills.
+        """
+        # All imports are by default relative to the root config file directory
+        config = os.path.abspath(config)
+        selector = cls(set(), os.path.dirname(config))
+        selector = cls._from_file(config, selector)
+
+        logger.debug("Selected skills: {}.".format(selector._imports))
+
+        return selector
+
+    @classmethod
+    def _from_path(cls, path: Text, skill_selector: "SkillSelector") -> "SkillSelector":
+        if os.path.isfile(path):
+            return cls._from_file(path, skill_selector)
+        elif os.path.isdir(path):
+            return cls._from_directory(path, skill_selector)
+        else:
+            logger.debug("No imports found. Importing everything.")
+            return cls.empty()
+
+    @classmethod
+    def _from_file(cls, path: Text, skill_selector: "SkillSelector") -> "SkillSelector":
+        path = os.path.abspath(path)
+        if os.path.exists(path) and data.is_config_file(path):
+            config = io_utils.read_yaml_file(path)
+
+            if isinstance(config, dict):
+                parent_directory = os.path.dirname(path)
+                return cls._from_dict(config, parent_directory, skill_selector)
+
+        return cls.empty()
+
+    @classmethod
+    def _from_dict(
+        cls, _dict: Dict, parent_directory: Text, skill_selector: "SkillSelector"
+    ) -> "SkillSelector":
+        imports = _dict.get("imports") or []
+        imports = {os.path.join(parent_directory, i) for i in imports}
+        # clean out relative paths
+        imports = {os.path.abspath(i) for i in imports}
+        import_candidates = [
+            p
+            for p in imports
+            if skill_selector.is_empty() or not skill_selector.is_imported(p)
+        ]
+        new = cls(imports, parent_directory)
+        skill_selector = skill_selector.merge(new)
+
+        # import config files from paths which have not been processed so far
+        for p in import_candidates:
+            other = cls._from_path(p, skill_selector)
+            skill_selector = skill_selector.merge(other)
+
+        return skill_selector
+
+    @classmethod
+    def _from_directory(
+        cls, path: Text, skill_selector: "SkillSelector"
+    ) -> "SkillSelector":
+        for parent, _, files in os.walk(path):
+            for file in files:
+                full_path = os.path.join(parent, file)
+
+                if data.is_config_file(full_path) and skill_selector.is_imported(
+                    full_path
+                ):
+                    skill_selector = cls._from_file(full_path, skill_selector)
+
+        return skill_selector
+
+    def merge(self, other: "SkillSelector") -> "SkillSelector":
+        imports = self._imports | {
+            i for i in other._imports if not self.is_imported(i) or self.is_empty()
```

Comment:
the whole naming of `empty` is very confusing. it should rather be something like `should_use_all()`

there should be three states: empty, some and all imports are selected

Reply:
renamed. What about the classfunction to create an empty selector? Empty means for me "no imports found, import everything you can"